### PR TITLE
DG and UG Update

### LINKED ISSUE #108: Update User Guide and Developer Guide Common Sections

### PR DESCRIPTION
**Pull request by Respirayson**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -278,142 +276,109 @@ Below is a class diagram describing the implementation of `Task` and its respect
...
 
-<box type="info" seamless>
+#### Design Considerations:
 
-**Note:** The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+**Aspect: How the students are stored to a consultation:**
 
-</box>
+* **Alternative 1 (Current choice):** Use Set<Person> to keep track of students in a consultation.
+    * Pros: Store only one of the same person, no duplicates.
```
Comment (raydenlim) [inline docs/DeveloperGuide.md]: Stores only 1 instance of a unique person, no duplicates

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -278,142 +276,109 @@ Below is a class diagram describing the implementation of `Task` and its respect
...
-**Note:** If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone AddressBook states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+**Aspect: Adding students to a new or existing consultation:**
 
-</box>
+* **Alternative 1 (Current choice):** The `AddToConsult` feature creates a new Consultation object with updated student list
+  * Pros: Defensive programming when entirely creating a new Consultation object without modifying previous object.
+  * Cons: Require additional checking to inform exception cases.
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
+* **Alternative 2:** `AddToConsult` directly manipulate the attribute students in a Consultation object.
```
Comment (raydenlim) [inline docs/DeveloperGuide.md]: Might want to use markdowns here: attribute `students`

**Review by wesho1107 — APPROVED** [review]
LGTM!

**Review by raydenlim — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (wesho1107) [reply docs/DeveloperGuide.md]: I agree

**Review by wesho1107 — APPROVED** [review]
Other than rephrasing, LGTM

**Pull request merged**